lib/honeybadger/insights/ecto.ex

```diff
@@ -62,6 +70,37 @@ defmodule Honeybadger.Insights.Ecto do
     meta
     |> Map.take([:query, :decode_time, :query_time, :queue_time, :source])
     |> Map.update!(:query, &obfuscate(&1, meta.repo.__adapter__()))
+    |> include_stacktrace(meta)
+  end
+
+  defp include_stacktrace(data, %{stacktrace: stacktrace}) do
+    if get_insights_config(:include_stacktrace, false) do
+      Map.put(data, :stacktrace, format_stacktrace(stacktrace))
+    else
+      data
+    end
+  end
+
+  defp include_stacktrace(data, _), do: data
+
+  defp format_stacktrace(stacktrace) do
+    stacktrace
+    |> Enum.map(&format_frame/1)
+  end
+
+  defp format_frame({module, function, arity, location}) do
+    if is_list(location) do
+      file = location[:file]
+      line = location[:line]
+
+      if file && line do
+        ["#{file}:#{line}", "#{inspect(module)}.#{function}/#{arity}"]
+      else
+        [nil, "#{inspect(module)}.#{function}/#{arity}"]
+      end
+    else
+      [nil, "#{inspect(module)}.#{function}/#{arity}"]
+    end
```

Comment:
Can remove some duplication and flatten out the conditional:

```suggestion
    position =
      if is_list(location) and Keyword.has_key?(location, :file) do
        "#{location[:file]}:#{location[:line]}"
      else
        nil
      end

    [position, Exception.format_mfa(module, function, arity)]
```

Reply:
Oooh, handy!